source/slang/slang-ir-redundancy-removal.cpp

```diff
@@ -159,26 +159,29 @@ bool removeRedundancyInFunc(IRGlobalValueWithCode* func)
     return result;
 }
 
-// Remove IR definitions from all [AvailableInDXIL] functions when compiling DXIL,
-// as these functions are already defined in the embedded precompiled DXIL library.
-void removeAvailableInDownstreamModuleDecorations(IRModule* module)
+// Remove IR definitions from all AvailableInDownstreamIR functions where the
+// languages match what we're currently targetting,  as these functions are
+// already defined in the embedded precompiled library.
+void removeAvailableInDownstreamModuleDecorations(CodeGenTarget target, IRModule* module)
 {
     List<IRInst*> toRemove;
     for (auto globalInst : module->getGlobalInsts())
     {
-        auto funcInst = as<IRFunc>(globalInst);
-        if (!funcInst)
+        if (auto funcInst = as<IRFunc>(globalInst))
         {
-            continue;
-        }
-        if (globalInst->findDecoration<IRAvailableInDXILDecoration>())
-        {
-            // Gut the function definition, turning it into a declaration
-            for (auto inst : funcInst->getChildren())
+            if (auto dec = globalInst->findDecoration<IRAvailableInDownstreamIRDecoration>())
             {
-                if (inst->getOp() == kIROp_Block)
+                if ((dec->getTarget()->getValue() == SLANG_DXIL && target == CodeGenTarget::HLSL) ||
```

Comment:
Can this be simplified to just `dec->getTarget()->getValue() == target`?

Reply:
The current code gen target is HLSL at this point, but the precompiled blob is DXIL.